### PR TITLE
Replace Travis.CI tests with GitHub action

### PR DESCRIPTION
**Pull request by peterhashair**
### Description:

Travis to github action

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: .github/workflows/plugin_tests.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    strategy:
+      fail-fast: false
+    steps:
+      - name: run tests
+        uses: matomo-org/github-action-tests@v1
+        with:
+          git-head: ${{ github.head_ref }}
+          git-sha: ${{ github.event.pull_request.head.sha }}
+          plugin-name: 'TasksTimetable'
+          php-version: '7.4'
```
Comment (sgiehl) [inline .github/workflows/plugin_tests.yml]: I think we used to run UI tests on the minimum supported PHP version only, as that helps to find compatibility issues with code not covered by other tests. So might be good to change that to `7.2.` again.

File: .github/workflows/plugin_tests.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+    steps:
+      - name: run tests
+        uses: matomo-org/github-action-tests@v1
+        with:
+          git-head: ${{ github.head_ref }}
+          git-sha: ${{ github.event.pull_request.head.sha }}
+          plugin-name: 'TasksTimetable'
```
Comment (sgiehl) [inline .github/workflows/plugin_tests.yml]: Here I would have expected, not only passing the head/sha for the plugin repo to test with, but also the Matomo branch/tag to test against. Before that was handled with `PIWIK_TEST_TARGET`, which was `4.x-dev` for this plugin.
But for other tests than UI tests that was also used with the minimum required Matomo version (tag).

File: .github/workflows/plugin_tests.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+name: Plugin TasksTimetable Tests
+
+on:
+  pull_request:
+  schedule:
+    - cron: '0 0 * * *'
```
Comment (sgiehl) [inline .github/workflows/plugin_tests.yml]: On travis we had the tests setup in a way, that they were triggered once a week. Don't think we need to run them daily...

**Review by sgiehl — DISMISSED** [review]

File: .github/workflows/plugin_tests.yml (hunk 3 above)
Comment (peterhashair) [reply .github/workflows/plugin_tests.yml]: removed

File: .github/workflows/plugin_tests.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    strategy:
+      fail-fast: false
+    steps:
+      - name: run tests
+        uses: matomo-org/github-action-tests@v1
+        with:
+          git-head: ${{ github.head_ref }}
+          git-sha: ${{ github.event.pull_request.head.sha }}
+          plugin-name: 'TasksTimetable'
+          php-version: '7.4'
```
Comment (peterhashair) [inline .github/workflows/plugin_tests.yml]:
```suggestion
          php-version: '7.2'
```

File: .github/workflows/plugin_tests.yml (hunk 1 above)
Comment (peterhashair) [reply .github/workflows/plugin_tests.yml]: change to 7.2

File: .github/workflows/plugin_tests.yml (hunk 2 above)
Comment (peterhashair) [reply .github/workflows/plugin_tests.yml]: added `test-target`

**Pull request merged**